### PR TITLE
simplify dearmor

### LINKED ISSUE #1197: Simplify `rnp_dearmor_source`
## Description

As per discussion in PR #1177

### PR DESCRIPTION
**Pull request by rrrooommmaaa**
Closes #1197

### CODE REVIEW

File: src/librepgp/stream-armor.cpp [hunk 1]
```diff
@@ -973,48 +973,35 @@ armored_dst_set_line_length(pgp_dest_t *dst, size_t llen)
 bool
 is_armored_source(pgp_source_t *src)
 {
-    uint8_t buf[128];
-    size_t  read = 0;
+    uint8_t buf[strlen(ST_ARMOR_BEGIN)];
```
Comment (rrrooommmaaa) [inline src/librepgp/stream-armor.cpp:976]: @ni4 you were saying previously that there might be EOL and other whitespace prepending `ST_ARMOR_BEGIN`, this is why `strstr` is used to determine the type of source? And this whitespace is limited by 100-something bytes? Is this a limitation by the protocol?

Comment (ni4) [reply src/librepgp/stream-armor.cpp:976]: @rrrooommmaaa Yep. This is not a protocol limitation but heuristic, allowing to work in most of the cases. 128 bytes are picked to be enough for most of the cases.

**Comment by ni4** [comment]
@rrrooommmaaa Would you have time to update this so we have less opened/abandoned PRs?

**Comment by ni4** [comment]
Closing this as was resolved via the 4a2d42d9591ecf7428a738247181c988f6fc0ff2

**Pull request closed without merging**